akka-docs-dev/rst/scala/stream-customize.rst

```diff
@@ -129,10 +135,144 @@ Finally, there are two methods available for convenience to complete the stage a
 * ``failStage(exception)`` is equivalent to failing all output ports and cancelling all input ports.
 
 
+In some cases it is inconvenient and error prone to react on the regular state machine events with the
+signal based API described above. For those cases there is a API which allows for a more imperative sequencing
+of actions which will greatly simplify some use cases at the cost of some extra allocations. The difference
+between the two APIs could be described as that the first one is signal driven from the outside, while this API
+is more active and drives it surroundings.
+
+The operations of this part of the :class:``GraphStage`` API are:
+
+* ``emit(out, elem)`` and ``emitMultiple(out, Iterable(elem1, elem2))`` replaces the ``OutHandler`` with s handler that emit
+  one or more elements when there is demand, and then reinstall the current handlers
+* ``read(in)(andThen)`` and ``readN(in, n)(andThen)`` replaces the handlers that will
```

Comment:
This now also has an onClose handler—reading without it didn’t make much sense, but before MergeSorted there was no use-case that exposed the problem.